pkg/virtual/syncer/coordination/helpers.go

```diff
@@ -0,0 +1,228 @@
+func NewDefaultSyncerViewManager[T Object]() SyncerViewManager[T] {
+	return NewSyncerViewManager[T](&transformations.SpecDiffTransformation{}, &transformations.DefaultSummarizingRules{})
+}
+
+func toUnstructured[T Object](obj T) (*unstructured.Unstructured, error) {
+	unstructured := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	raw, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		return nil, err
+	}
+	unstructured.Object = raw
+	return unstructured, nil
+}
+
+func fromUnstructured[T Object](unstr *unstructured.Unstructured) (T, error) {
+	var obj T
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstr.Object, &obj)
+	if err != nil {
+		return obj, err
+	}
+	return obj, nil
+}
+
+func (svm *syncerViewManager[T]) GetFilteredSyncerViews(ctx context.Context, gvr schema.GroupVersionResource, upstreamResource T, keepSyncTarget func(key string) bool) (map[string]T, error) {
+	syncerViews := make(map[string]T)
+	for name := range upstreamResource.GetAnnotations() {
+		if strings.HasPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix) {
+			syncTargetKey := strings.TrimPrefix(name, v1alpha1.InternalSyncerViewAnnotationPrefix)
+			if keepSyncTarget(syncTargetKey) {
+				var unstrResource *unstructured.Unstructured
+				if unstructured, isUnstructured := Object(upstreamResource).(*unstructured.Unstructured); isUnstructured {
+					unstrResource = unstructured
+				} else {
+					unstructured, err := toUnstructured[T](upstreamResource)
+					if err != nil {
+						return nil, err
+					}
+					unstrResource = unstructured
+				}
+				if unstrSyncerView, err := svm.SyncerResourceTransformer.AfterRead(unimplemented, syncercontext.WithSyncTargetKey(ctx, syncTargetKey), gvr, unstrResource, nil); err != nil {
+					return nil, err
+				} else {
+					var syncerView T
+					if typedSyncerView, isTyped := Object(unstrSyncerView).(T); isTyped {
+						syncerView = typedSyncerView
+					} else {
+						if typedSyncerView, err = fromUnstructured[T](unstrSyncerView); err != nil {
+							return nil, err
+						} else {
+							syncerView = typedSyncerView
+						}
+					}
+					syncerViews[syncTargetKey] = syncerView
+				}
+			}
+		}
+	}
+	return syncerViews, nil
+}
+
+func (svm *syncerViewManager[T]) GetSyncerViews(ctx context.Context, gvr schema.GroupVersionResource, upstreamResource T) (map[string]T, error) {
+	return svm.GetFilteredSyncerViews(ctx, gvr, upstreamResource, func(key string) bool {
+		return true
+	})
+}
+
+func (svm *syncerViewManager[T]) GetSyncerView(ctx context.Context, gvr schema.GroupVersionResource, upstreamResource T, syncTargetKey string) (T, error) {
+	if views, err := svm.GetFilteredSyncerViews(ctx, gvr, upstreamResource, func(key string) bool {
+		return syncTargetKey == key
+	}); err != nil {
+		var zeroVal T
+		return zeroVal, err
+	} else {
+		return views[syncTargetKey], nil
+	}
+}
+
+var unimplemented dynamic.ResourceInterface = &unimplementedDynamicClient{}
```

Comment:
What is this? Needed for tests? Then move it to tests out of production code.

Reply:
Removed (did another way)